pkg/trace/config/config.go

```diff
@@ -87,18 +87,9 @@ type ObfuscationConfig struct {
 
 	// CreditCards holds the configuration for obfuscating credit cards.
 	CreditCards CreditCardsConfig `mapstructure:"credit_cards"`
-}
 
-// AppSecConfig ...
-type AppSecConfig struct {
-	// Enabled reports whether AppSec is enabled.
-	Enabled bool
-	// MaxPayloadSize ...
-	MaxPayloadSize int64
-	// APIKey ...
-	APIKey string
-	// DDURL ...
-	DDURL string
+	// AppSec holds the obfuscation settings for the AppSec meta value.
+	AppSec AppSecObfuscationConfig
```

Comment:
@gbbr: since your latest `pkg/trace/config` changes, I no longer see how could I load the Datadog configuration values from here when you create this object with the default values.

I currently defined them in `pkg/config/appsec.go`. Should I add this package to your go.mod or should I move or copy the default config values?

Reply:
Ok, so I fixed the overall situation in https://github.com/DataDog/datadog-agent/pull/11324/commits/149f8d3784a851bfaedadf5ae14b46f1904a6b7d where:
1. The default configuration values were moved into `pkg/trace/config` and reused in `pkg/config` from there.
2. My compile regexp helper function was moved into the obfuscator pacakge and reused from `pkg/trace/config` and `cmd/trace-agent` to compile the configuration values.

Overall, this looks way better.